ddtrace/tracer/rules_sampler.go

```diff
@@ -38,9 +37,9 @@ type rulesSampler struct {
 // Rules are split between trace and single span sampling rules according to their type.
 // Such rules are user-defined through environment variable or WithSamplingRules option.
 // Invalid rules or environment variable values are tolerated, by logging warnings and then ignoring them.
-func newRulesSampler(traceRules, spanRules []SamplingRule, traceSampleRate float64) *rulesSampler {
+func newRulesSampler(traceRules, spanRules []SamplingRule, traceSampleRate, rateLimit float64) *rulesSampler {
```

Comment:
`rateLimit` does not indicate what rate it is limiting in this context... Perhaps mention it in the documentation? It's unclear how it differs from `traceSampleRate`.

Also, I personally prefer to type each variable.

```suggestion
func newRulesSampler(traceRules, spanRules []SamplingRule, traceSampleRate float64, rateLimit float64) *rulesSampler {
```

Reply:
I tend to favour the implicit type for contiguous arguments because I'm used to it.